README.md

```diff
@@ -42,7 +42,346 @@ Or install it yourself as:
+  to 'noreply@recipient.com'
+  subject 'Welcome'
+end
+
+CustomDelivery = {
+  :address              => "smtp.gmail.com",
+  :port                 => 587,
+  :domain               => "localhost:8000",
+  :user_name            => "noreply@sender.com",
+  :password             => "password",
+  :authentication       => "plain",
+  :enable_starttls_auto => true
+}
+
+Lotus::Mailer.configuration do
+  delivery_method CustomDelivery
+end
+
+Lotus::Mailer.load!
+InvoiceMailer.deliver
+```
+
+### Locals
+
+The set of objects passed in the `deliver` call are called `locals` and are used to render templates that use objects with the same symbol.
+
+```ruby
+require lotus/mailer
+
+class User
+  def initialize(name, username)
+    @name = name
+    @username = username
+  end
+
+  attr_reader :name, :username
+end
+
+luca = User.new('Luca', 'jodosha')
+InvoiceMailer.deliver(user:luca)
+```
+
+The corresponding `erb` file:
+
+```erb
+Hello <%= user.name %>!
+```
+
+### Scope
+
+All the methods defined in the mailer are accessible from the template:
+
+```ruby
+require lotus/mailer
+
+class InvoiceMailer
+  include Lotus::Mailer
+  
+  from 'noreply@sender.com'
+  to 'noreply@recipient.com'
+  subject 'Welcome'
+  
+  def greeting
+    'Ahoy'
+  end
+end
+```
+
+```erb
+<h2><%= greeting %></h2>
+```
+
+### Template
+The template file must be located under the relevant `root` and must match the class name in camel case:
+
+```ruby
+puts Lotus::Mailer.configuration.root # => #<Pathname:app/templates>
+InvoiceMailer.templates               # => "{:html => templateObject1; :txt => templateObject2}"
+templateObject1.file                  # => "root/invoice_mailer.html.erb"
+templateObject2.file                  # => "root/invoice_mailer.txt.erb"
+```
+
+Each mailer can specify a different template for a specific format:
+
+```ruby
+module Articles
+  class Create
+    include Lotus::View
+
+    template :haml, 'invoice.haml.erb'
+  end
+end
+
+InvoiceMailer.templates[:haml].file  # => "root/invoice.haml.erb"
+```
+
+### Engines
+
+The builtin rendering engine is [ERb](http://en.wikipedia.org/wiki/ERuby).
+
+### Root
+
+Template lookup is performed under the `Lotus::Mailer.configuration.root` directory. You can specify a different path on a per mailer basis:
+
+```ruby
+class MailerWithDifferentRoot
+  include Lotus::Mailer
+
+  root 'path/to/root'
+end
+```
+
+### Configuration
+
+__Lotus::Mailer__ can be configured with a DSL that determines its behavior.
+It supports a few options:
+
+```ruby
+require 'lotus/mailer'
+
+Lotus::Maler.configure do
+  # Set the root path where to search for templates
+  # Argument: String, Pathname, #to_pathname, defaults to the current directory
+  #
+  root '/path/to/root'
+
+  # Set the Ruby namespace where to lookup for mailers
+  # Argument: Class, Module, String, defaults to Object
+  #
+  namespace 'MyApp::Mailer'
+```
+
+All those global configurations can be overwritten at a finer grained level:
```

Comment:
Should you mention that configurations are lazy-loaded and calling `Lotus::Mailer.load!` is required?